File: jbake-core/src/main/java/org/jbake/app/DBUtil.java
package org.jbake.app;

import org.jbake.app.configuration.JBakeConfiguration;
import org.jbake.model.DocumentModel;

import java.util.ArrayList;

public class DBUtil {
    private static ContentStore contentStore;

    @Deprecated
    public static ContentStore createDataStore(final String type, String name) {
        if (contentStore == null) {
            contentStore = new ContentStore();
        }
        return contentStore;
    }

    @Deprecated
    public static void updateSchema(final ContentStore db) {
        // no-op
    }

    public static ContentStore createDataStore(JBakeConfiguration configuration) {
        if (contentStore == null) {
            contentStore = new ContentStore();
        }

        return contentStore;
    }

    public static void closeDataStore() {
        contentStore = null;
    }

    /**
     * Converts a list entry into a String array.
     *
     * @param entry Entry input to be converted
     * @return input entry as String[]
     */
    @SuppressWarnings("unchecked")
    public static String[] toStringArray(Object entry) {
        if (entry instanceof String[]) {
            return (String[]) entry;
        } else if (entry instanceof ArrayList) {
            ArrayList<String> list = (ArrayList<String>) entry;
            return list.toArray(new String[0]);
        }
        return new String[0];
    }

}

File: jbake-core/src/main/java/org/jbake/app/DBUtil.kt
package org.jbake.app

import com.orientechnologies.orient.core.db.record.OTrackedList
import com.orientechnologies.orient.core.sql.executor.OResult
import org.jbake.app.configuration.JBakeConfiguration
import org.jbake.model.DocumentModel

object DBUtil {
    private var contentStore: ContentStore? = null

    @Deprecated("")
    fun createDataStore(type: String, name: String): ContentStore {
        if (contentStore == null) {
            contentStore = ContentStore(type, name)
        }
        return contentStore!!
    }

    @Deprecated("")
    fun updateSchema(db: ContentStore) {
        db.updateSchema()
    }

    @JvmStatic
    fun createDataStore(configuration: JBakeConfiguration): ContentStore {
        return contentStore ?: ContentStore(configuration.databaseStore, configuration.databasePath)
            .also { contentStore = it }
    }

    fun closeDataStore() {
        contentStore = null
    }

    fun documentToModel(doc: OResult): DocumentModel {
        val result = DocumentModel()
        doc.propertyNames.forEach { key ->
            result[key] = doc.getProperty(key)
        }
        return result
    }

    /**
     * Converts a DB list into a String array
     *
     * @param entry Entry input to be converted
     * @return input entry as String[]
     */
    fun toStringArray(entry: Any): Array<String> = when (entry) {
        is Array<*> -> entry as? Array<String> ?: emptyArray()
        is OTrackedList<*> -> (entry as? OTrackedList<String>)?.toTypedArray() ?: emptyArray()
        is ArrayList<*> -> (entry as? ArrayList<String>)?.toTypedArray() ?: emptyArray()
        else -> arrayOf()
    }
}
